src/components/CanvasContainer/CanvasContainer.tsx

```diff
@@ -9,13 +9,16 @@ interface CanvasContainerProps {
 export const CanvasContainer: React.FC<CanvasContainerProps> = ({ uploadedFileData }) => {
   const canvasContainerRef = React.useRef(document.createElement("div"));
 
-  React.useEffect(() => {
+  useEffect(() => {
     const canvas = renderer.init();
     if (canvasContainerRef.current) {
       canvasContainerRef.current.appendChild(canvas);
-      canvasContainerRef.current.style.backgroundImage = "url(" + uploadedFileData + ")";
     }
-  }, [canvasContainerRef, uploadedFileData]);
+  }, [canvasContainerRef]);
+
+  useEffect(() => {
```

Comment:
Why did you move this to another useEffect function?

Reply:
Because when we get to the measurement Page, we set uploadedFileData to "" to remove the photo from the brackground. This change triggered the useEffect that reinitialized the renderer uselessly and re-appended the canvas as a child of the canvasContainerRef unnecessarily, which made getting access to the canvas later weird and buggy

Reply:
I see, would you please add that explanation as a comment?

Reply:
done!